Include trip shape on the VPP map

skyqrose opened the pull request:
Asana Task: [Include trip shape on the VPP map](https://app.asana.com/0/1148853526253426/1134108215297992)

<img width="333" alt="Screen Shot 2020-01-15 at 15 05 45" src="https://user-images.githubusercontent.com/23065557/72467651-5e574600-37a9-11ea-9d5a-803c496921be.png">

* Save shapes for non shuttle routes
* Adds an endpoint for getting a single trip's shape.
* Calls `useTripShape` from the VPP.

The commits are well separated if that helps.

The result is stored in the VPP component, which means we need to refetch it every time you open a VPP. Do you think it's worth calling `useTripShape` from the `AppStateWrapper` component instead so it can be cached?

Previous version with all the shapes for the route, instead of just the one shape on the vehicle's trip: #412

skyqrose commented on assets/src/api.ts:
```diff
@@ -59,10 +66,21 @@ export const fetchRoutes = (): Promise<Route[]> =>
+    .then(({ data: shape }: { data: Shape | null }) => shape)
     .catch(error => {
       // tslint:disable-next-line: no-console
       console.error(error)
```
I'd rather put more thought into fixing this for all the api calls at once, rather than one by one for new calls. Followup task: [Properly handle errors in api calls.](https://app.asana.com/0/1152340551558956/1158195594242473)

arkadyan replied:
Fixing the other cases is a great followup task, but why not just do this one correctly right now?

skyqrose replied:
Because I don't want to add a special case for this function, and then immediately refactor it out.

Default values for all api calls where it makes sense are ready to go in #424 (built on top of this PR).

skyqrose replied:
I know we're having a discussion on #424 so it won't be ready to go immediately after this, but I don't think this PR needs to wait for it. The reason to return the default now in this PR would be so that it's in a better state for the long term if we don't follow up, but with that PR open already, I don't think that's a big risk.

skyqrose commented:
Ready for another review.

I added tests, and changed `useTripShape` and `useRouteShapes` to both return a list of only the loaded shapes, so the caller doesn't have to process them.

skyqrose commented:
Note that you'll need to `mix cache.clean` locally to see this.
